IntegrationTests/Services/AWSS3IntegrationTests/S3SigV4ATests.swift

```diff
@@ -199,4 +146,59 @@ class S3SigV4ATests: S3XCTestCase {
         )
         XCTAssertEqual(response.keyCount, 1)
     }
+
+    // MARK: - HELPER FUNCTIONS
+
+    // Check if MRAP exists by checking names of existing MRAPs against integ-test MRAP prefix.
+    private func checkIfMRAPExists() async throws -> S3ControlClientTypes.MultiRegionAccessPointReport? {
+        let mraps = try await s3ControlClient.listMultiRegionAccessPoints(input: ListMultiRegionAccessPointsInput(accountId: accountId)).accessPoints
+        return mraps?.first { $0.name?.hasPrefix(mrapNamePrefix) ?? false }
+    }
+
+    private func constructMRAPConfig() {
+        let publicAccessBlock = S3ControlClientTypes.PublicAccessBlockConfiguration(
+            blockPublicAcls: false,
+            blockPublicPolicy: false,
+            ignorePublicAcls: false,
+            restrictPublicBuckets: false
+        )
+        let regions = [S3ControlClientTypes.Region(bucket: bucketName, bucketAccountId: accountId)]
+        mrapConfig = S3ControlClientTypes.CreateMultiRegionAccessPointInput(
+            name: mrapName,
+            publicAccessBlock: publicAccessBlock,
+            regions: regions
+        )
+    }
+
+    private func constructMRAP() async throws {
+        // Construct MRAP config to use
+        constructMRAPConfig()
+        // Create S3 Multi-Region Access Point (MRAP)
+        let createMRAPInput = CreateMultiRegionAccessPointInput(
+            accountId: accountId,
+            clientToken: UUID().uuidString.split(separator: "-").first!.lowercased(),
+            details: mrapConfig
+        )
+        _ = try await s3ControlClient.createMultiRegionAccessPoint(input: createMRAPInput)
+
+        // Wait until MRAP creation finishes
+        var status: S3ControlClientTypes.MultiRegionAccessPointStatus? = .creating
+        repeat {
+            let seconds = 20.0
+            try await Task.sleep(nanoseconds: UInt64(seconds * Double(NSEC_PER_SEC)))
+
+            status = try await s3ControlClient.getMultiRegionAccessPoint(input: GetMultiRegionAccessPointInput(
+                accountId: accountId,
+                name: mrapName
+            )).accessPoint?.status
+        } while status == .creating
```

Comment:
Do we need to have some sort of timeout mechanism for this `repeat` loop?

Reply:
Added a simple polling limit of 20 with poll count incrementing by 1 each time.